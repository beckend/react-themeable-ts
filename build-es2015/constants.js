"use strict";
exports.THEME_PROVIDER_CONTEXT_KEY = 'reactThemeable';

//# sourceMappingURL=constants.js.map
