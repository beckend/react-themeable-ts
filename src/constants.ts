export const THEME_PROVIDER_CONTEXT_KEY = 'reactThemeable';
