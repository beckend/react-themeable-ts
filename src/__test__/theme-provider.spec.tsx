/* tslint:disable: variable-name */
import * as React from 'react';
import {
  ThemeProvider,
  IThemeProviderContext,
} from '../theme-provider';
import { THEME_PROVIDER_CONTEXT_KEY } from '../constants';
import {
  themeDecorator,
  IHOCDefaultNoThemeProps,
} from '../hoc';

const renderer = require('react-test-renderer');

describe('theme-provider context tests', () => {
  it('basic pass all props', () => {
    const ComponentWithContext1: React.SFC<{}> = (props, context: IThemeProviderContext) => {
      return (
        <div>
          {context[THEME_PROVIDER_CONTEXT_KEY].mystr}
        </div>
      );
    };

    ComponentWithContext1.contextTypes = {
      [THEME_PROVIDER_CONTEXT_KEY]: React.PropTypes.object.isRequired,
    };

    const component = renderer.create(
      <ThemeProvider
        reactThemeable={{
          mystr: 'context string',
        }}
      >
        <ComponentWithContext1 />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('theme-provider context with HOC test', () => {
  it('should get the correct context key and render theme', () => {
    interface IComponentWithContext2NoHOCProps {}
    interface IComponentWithContext2Props extends IComponentWithContext2NoHOCProps, IHOCDefaultNoThemeProps {}
    const ComponentWithContext2: React.SFC<IComponentWithContext2Props> = (props, context: IThemeProviderContext) => {
      const { t } = props;
      return (
        <div {...t('root')}>
          <span {...t('span1')}>
            My test SFC component
          </span>
          <p {...t('p1')}>
            My test SFC component
          </p>
        </div>
      );
    };

    ComponentWithContext2.contextTypes = {
      [THEME_PROVIDER_CONTEXT_KEY]: React.PropTypes.object.isRequired,
    };

    const decoratorFn = themeDecorator({
      context: true,
      contextPath: [THEME_PROVIDER_CONTEXT_KEY, 'ComponentWithContext2'],
    });
    const MyDecoratedComponent: React.SFC<IComponentWithContext2NoHOCProps> = decoratorFn(ComponentWithContext2);
    const component = renderer.create(
      <ThemeProvider
        reactThemeable={{
          ComponentWithContext2: {
            root: 'class-root',
            span1: 'class-span1',
            p1: 'class-p1',
          },
        }}
      >
        <MyDecoratedComponent />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
