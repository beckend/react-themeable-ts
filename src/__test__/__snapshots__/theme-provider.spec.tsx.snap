exports[`theme-provider context tests basic pass all props 1`] = `
<div>
  context string
</div>
`;

exports[`theme-provider context with HOC test should get the correct context key and render theme 1`] = `
<div
  className="class-root">
  <span
    className="class-span1">
    My test SFC component
  </span>
  <p
    className="class-p1">
    My test SFC component
  </p>
</div>
`;
